src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/inference/CpxVariantReInterpreterSpark.java

```diff
@@ -0,0 +1,856 @@
+                    final int altSeqLength = vc.getAttributeAsString(GATKSVVCFConstants.SEQ_ALT_HAPLOTYPE, "").length() - 1;
+                    final VariantContextBuilder vcBuilderForIns = new VariantContextBuilder()
+                            .chr(vc.getContig()).start(vc.getStart()).stop(vc.getStart())
+                            .id(makeIDForIns(vc.getContig(), vc.getStart(), vc.getStart()))
+                            .alleles(Arrays.asList(anchorBaseRefAllele, altSymbAlleleIns))
+                            .attribute(GATKSVVCFConstants.SVTYPE, SimpleSVType.TYPES.INS.name())
+                            .attribute(GATKSVVCFConstants.SVLEN, altSeqLength)
+                            .attribute(EVENT_KEY, vc.getID())
+                            .attribute(GATKSVVCFConstants.CONTIG_NAMES, makeSureAttributeIsList(vc, GATKSVVCFConstants.CONTIG_NAMES));
+                    return vcBuilderForIns.make();
+                });
+
+        return Stream.concat(
+                reInterpreted,
+                oneSegmentCalls.flatMap(vc -> CpxVariantReInterpreterSpark.workerForOneSegmentCalls(vc, reference))
+        ).collect(Collectors.toList());
+    }
+
+    /**
+     * Depending on if the ref segment is present in alt arrangement, logic as follows:
+     * <ul>
+     *     <li>
+     *         ref segment is absent
+     *         <ul>
+     *             <li> present inverted -> INV (if size > 49) and INS (if new material > 49) </li>
+     *             <li> not at all       -> DEL (if size > 49) and INS (if new material > 49) </li>
+     *         </ul>
+     *     </li>
+     *
+     *     <li>
+     *         ref segment is present
+     *         <ul>
+     *             <li> there's >=1 inverted presence -> INV (if size > 49) and INS (if new material subtracting the INV > 49) </li>
+     *             <li> there's no  inverted presence -> INS (if new material > 49) </li>
+     *         </ul>
+     *     </li>
+     * </ul>
+     */
+    private static Stream<VariantContext> workerForOneSegmentCalls(final VariantContext vc, final ReferenceMultiSource reference) {
+
+        final List<String> evidenceContigs = makeSureAttributeIsList(vc, GATKSVVCFConstants.CONTIG_NAMES);
+        final SimpleInterval refSegment =
+                new SimpleInterval(vc.getAttributeAsString(GATKSVVCFConstants.CPX_SV_REF_SEGMENTS, ""));
+        final int segmentSize = refSegment.size();
+
+        final int altSeqLength = vc.getAttributeAsString(GATKSVVCFConstants.SEQ_ALT_HAPLOTYPE, "").length() - 1;
+
+        final List<String> altArrangement = makeSureAttributeIsList(vc, GATKSVVCFConstants.CPX_EVENT_ALT_ARRANGEMENTS);
+
+        final List<VariantContext> result = new ArrayList<>();
+        final Allele anchorBaseRefAllele = Allele.create(vc.getAlleles().get(0).getBases()[0], true);
+        final Allele fatInsertionRefAllele;
```

Comment:
Copying the comments here:
* anchor base is defined per-VCF spec, that is, for DEL and INS variants, (see 1.4.1#REF version 4.2), the reference base at the position pointed to by POS, basically, for DEL, the reference bases immediately following POS are deleted (up to and including the END base), for INS, the INSSEQ are inserted immediately after POS; variants like INV are not required to be reported this way, so there's some ambiguity
* "fat" insertions exist because sometimes we have micro deletions surrounding the insertion breakpoint, so here the strategy is to report them as "fat", i.e. the anchor base and micro-deleted bases are reported in REF.